:sparkles: Add `animate me` behavior

miketheman opened the pull request:
The Google Custom Search API allows for a parameter to be passed to
specify the returned fileType.
The `imgType` does not currently support animated as a parameter, so we
can submit with `fileType:gif` to get animated images.
- Pop the query params out to its own Hash, so we can conditionally add to
  it before performing the `.get()`
- Add an input paramter to `fetch()` to control behavior
- Add a new route & method to control input to `.fetch()`

Closes #7 

---

@jimmycuadra This was how I was able to implement the desired behavior - and testing with real keys, I can confirm that it works as intended.

I'm not 100% sure this is the best approach, but I couldn't figure out a way to introspect the `command` to determine if the behavior should be image or animate.
The `fetch_anim()` method also felt a bit odd, for the same reason - totally open to other ideas/approaches for this.

miketheman committed:
:sparkles: Add `animate me` behavior

The Google Custom Search API allows for a parameter to be passed to
specify the returned fileType.
The `imgType` does not currently support animated as a parameter, so we
can submit with `fileType:gif` to get animated images.

- Pop the query params out to its own Hash, so we can conditionally add to
  it before performing the `.get()`
- Add an input paramter to `fetch()` to control behavior
- Add a new route & method to control input to `.fetch()`

Closes #7

jimmycuadra commented on lib/lita/handlers/google_images.rb:
```diff
@@ -21,11 +21,18 @@ class GoogleImages < Handler
+      })
+
+      def fetch_anim(response)
+        fetch(response, animated = true)
```
I think this should just be `fetch(response, true)`. (Same for the call in `fetch` below.)

jimmycuadra commented on lib/lita/handlers/google_images.rb:
```diff
@@ -34,6 +41,13 @@ def fetch(response)
           key: config.google_cse_key
+        }
+
+        query_params["fileType"] = "gif" if animated
```
Does this really guarantee the image is animated? Not all gifs are animated.

miketheman committed:
Update animating params and interface

Pass multiple secret-sauce invocations of query parameters that raise
the probability of a returned image actually be animated.

miketheman commented:
@jimmycuadra Thanks for the feedback - incorporated.
The "secret" parameters are lifted from https://github.com/hubot-scripts/hubot-google-images/pull/15

jimmycuadra commented:
https://github.com/jimmycuadra/lita-google-images/releases/tag/v3.1.0
